app/controllers/events_controller.rb

```diff
@@ -8,12 +8,28 @@ def search
     @event_list = Event.all
 
     if (params[:event])
-      @event         = Event.find_by(event_name_underscore: params[:event])
-      @event_rsvps   = @event.rsvps.order(status: :desc, updated_at: :desc)
-      @attending     = @event.rsvps.where(status: "Yes").length
-      @not_attending = @event.rsvps.where(status: "No").length
-      @no_reply      = @event.rsvps.where(status: "--").length
+      if (params[:event] == "messages")
```

Comment:
I think you can remove the `if (params[:event])` and `end` on lines 10 and 20 and change 11-13 to be 
```ruby
if params[:event] == "messages"
  @family_messages = gather_messages_by_family
elsif params[:event]
```
That would get rid of the double loop